magento/ece-tools
# MCLOUD-5706: Add support of new ES option to ECE-Tools

### Description

MC-20710 introduces a new ES option which will configure ES, so it should not be configured via env.php any more.

**AC**

- New ES option support is added
- MySQL for search engine is deprecated with warning

### Fixed Issues (if relevant)

1. https://jira.corp.magento.com/browse/MCLOUD-5706

### Manual testing scenarios

1. Use next git branches in `.magento.env.yaml`:
```
    DEPLOYED_MAGENTO_VERSION_FROM_GIT: '2.4.0'
    DEPLOY_FROM_GIT_OPTIONS:
      clear_magento_module_requirements: true
      repositories:
        ce:
          branch: es-cloud-test-chris
          repo: https://<KEY>@github.com/magento-honey-badgers/magento2ce.git
        ee:
          branch: es-cloud-test-chris
          repo: https://<KEY>@github.com/magento-honey-badgers/magento2ee.git
        sample-data-ce:
          branch: 2.4-develop
          repo: https://<KEY>@github.com/magento/magento2-sample-data.git
        sample-data-ee:
          branch: 2.4-develop
          repo: https://<KEY>@github.com/magento/magento2-sample-data-ee.git
```
2. Deploy Magento
3. Verify that storefront works as expected
4. Remove `elasticsearch` from relationships
5. Deploy Magento
6. Deployment fails with appropriate error

### Contribution checklist
